src/index.ts

```diff
@@ -0,0 +1,173 @@
+export const SECONDS_PER_HOUR = 60 * 60;
+/**
+ * In milliseconds
+ */
+export const SECONDS = 1000;
+/**
+* In milliseconds
+*/
+export const MINUTES = 60 * SECONDS;
+/**
+* In milliseconds
+*/
+export const HOURS = 60 * MINUTES;
+/**
+* In milliseconds
+*/
+export const DAYS = 24 * HOURS;
+
+/**
+ * Ensure an env var exists and return the contents if it does
+ */
+export const requiredVar = (varName: string): string => {
+	const s = optionalVar(varName);
+	if (s == null) {
+		throw new Error(`Missing environment variable: ${varName}`);
+	}
+	return s;
+};
+
+/**
+ * Fetches the contents of an env var if it exists and is not empty, otherwise returning the specified default value,
+ * this means that if no default is specified then `''` will be normalized to `undefined`
+ */
+export function optionalVar<R>(varName: string, defaultValue: R): string | R;
+export function optionalVar<R>(
+	varName: string,
+	defaultValue?: undefined,
+): string | undefined;
+export function optionalVar<R>(
+	varName: string,
+	defaultValue?: R,
+): string | R | undefined {
+	const s = process.env[varName];
+	if (s == null || s === '') {
+		return defaultValue;
+	}
+	return s;
+}
+
+/**
+ * Checks that a string is a valid int and returns the number if so
+ */
+export const checkInt = (s: string): number | undefined => {
+	s = s.trim();
+	if (/[^0-9]/.test(s)) {
+		return;
+	}
+	const i = parseInt(s, 10);
+	if (!Number.isFinite(i)) {
+		return;
+	}
+	return i;
+};
```

Comment:
This is checkUnsignedInt negative values are resulting in undefined.
Is this intended, that a '-10' shouldn't be parsed?

Reply:
Negatives should be fine, yes